crates/core_simd/src/ops.rs

```diff
@@ -144,8 +144,15 @@ macro_rules! impl_op {
             impl<const LANES: usize> core::ops::Neg for crate::$type<LANES> {
                 type Output = Self;
                 fn neg(self) -> Self::Output {
-                    Self::splat(0) - self
-                }
+                // TODO probs a smarter way to do this too
+                    if AsRef::<[$scalar]>::as_ref(&self)
+                        .iter()
+                        .any(|x| *x == <$scalar>::MIN)
+                        {
+                            panic!("neg(MIN) for integers is Undefined Behaviour");
```

Comment:
We might as well match the scalar panic messaging.

>thread 'main' panicked at 'attempt to negate with overflow', src/main.rs:3:24
so "attempt to negate with overflow" would be the preferred messaging here.

Reply:
Being consistent makes sense.
However, this is the only value that triggers the behavior - wouldn't it make sense to have the error say that? Or we just assume the user is aware of the edge case?
I'm in favor either way.

Reply:
I think the phrase used by rustc has an advantage in that it is "non-technical", it uses no special terminology from Rust (like function terminology or referencing a constant in std that in a given user's code might have been shadowed) to express the value from the set and there are several other ops which raise the "attempt to ${op} with overflow" pattern, making it easier to comprehend if it appears beside similar errors that don't involve Neg specifically.

Reply:
Is it actually accurate that it's UB anyway? If we were going to debug_assert instead my assumption would be that it's not.

Reply:
Well, we thought that it wasn't, but then we checked and it's apparently not, or at least rustc doesn't care. :^)

Reply:
Yeah, I originally mistakenly said it was UB (lumped it in with `div` and `rem` overflow)